nextflow_schema.json

```diff
@@ -13,6 +13,10 @@
             "properties": {
                 "input": {
                     "type": "string",
+                    "format": "file-path",
```

Comment:
Do we need to have these additional fields? i.e. `format`, `mimetype`, `pattern` ? Some pipelines are set-up to take multiple forms of `--input` that may break these definitions. Are we likely to need these fields for other stuff or can we assume that if a parameter in the schema has its own schema then it's a samplesheet?

Reply:
`"format": "file-path"` is the most important one, as that's the one that we need to tell GUIs etc that it's a file and not just an arbitrary string.

`mimetype` is not needed, that is to help a GUI pick a relevant editor (eg. a spreadsheet rather than a YAML code editor), but could potentially drop this in favour of just looking at the file extension if we want. I was thinking of Tower Datasets where you are creating the file through the UI, so the UI decides the filename - so can't autodetect extension. If you check the `public_data_ids` one I set it to `text/plain` so you could imagine that giving just a normal text box.

`pattern` is nothing new, I just figured that you may as well have it in the schema as it's already supported by everything we have. Means you can get rid of the filename checks in the pipeline code.

For other pipelines, you can basically pick whatever you want. For example in this schema I omitted most of these fields for the inputs that can be either a `.tar.gz` file or a directory - those are set to `path` instead of `file-path` so that they will accept either a directory or a file, and have no filename extension checking and no mimetype (and no schema).

Basically - I think that this is a useful feature beyond just samplesheets. So don't want to rely on schema annotations of sample sheets. We can also use schema to validate anything really. Could write a JSON schema to validate a BED file if you wanted for example...